go/packages/slslambda/basicspan.go

```diff
@@ -0,0 +1,120 @@
+package slslambda
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws"
+	"go.buf.build/protocolbuffers/go/serverless/sdk-schema/serverless/instrumentation/tags/v1"
+	"go.buf.build/protocolbuffers/go/serverless/sdk-schema/serverless/instrumentation/v1"
+	"sync"
+	"time"
+)
+
+type basicSpan struct {
+	name       string
+	start      time.Time
+	end        time.Time
+	children   []span
+	customTags map[string]string
+	errors     []errorEvent
+	warnings   []warningEvent
+
+	// Mutex is needed because the consumer may add data to the span from multiple goroutines.
+	sync.Mutex
+}
+
+func (s *basicSpan) Span() *basicSpan {
+	return s
+}
+
+func newSpan(name string) *basicSpan {
+	return &basicSpan{name: name, start: time.Now(), customTags: map[string]string{}}
+}
+
+func newSpanWithStartTime(name string, start time.Time) *basicSpan {
+	return &basicSpan{name: name, start: start, customTags: map[string]string{}}
+}
+
+func (s *basicSpan) Close() {
+	s.Lock()
+	defer s.Unlock()
+	if s.end.IsZero() {
+		s.end = time.Now()
+		for _, child := range s.children {
+			child.Close()
+		}
+	}
+}
+
+func (s *basicSpan) ToProto(traceID, spanID, parentSpanID []byte, requestID string, tags tags) *instrumentationv1.Span {
+	return &instrumentationv1.Span{
+		Id:                spanID,
+		TraceId:           traceID,
+		ParentSpanId:      parentSpanID,
+		Name:              s.name,
+		StartTimeUnixNano: uint64(s.start.UnixNano()),
+		EndTimeUnixNano:   uint64(s.end.UnixNano()),
+		CustomTags:        convertCustomTags(s.customTags),
+		Tags: &tagsv1.Tags{
+			OrgId: (*string)(&tags.OrganizationID),
+			Aws: &tagsv1.AwsTags{
```

Comment:
These tags should only be attached to the root span. We shouldn't attach them to every span. I think at the moment for custom spans the only tag that will be set is `orgId`

Reply:
done